src/core/ready-no-deferred.js

```diff
@@ -0,0 +1,109 @@
+define( [
+	"../core",
+	"../var/document"
+], function( jQuery, document ) {
+
+var readyCallbacks = [],
+	readyFiring = false,
+	whenReady = function( fn ) {
+		readyCallbacks.push( fn );
+	},
+	executeReady = function( fn ) {
+
+		// Prevent errors from freezing future callback execution (gh-1823)
+		// Not backwards-compatible as this does not execute sync
+		window.setTimeout( function() {
+			fn.call( document, jQuery );
```

Comment:
@gibson042 The change wouldn't just affect `jQuery.ready.then`, and I'm familiar with the spec on Promises. Like you said, the document context is for back-compat, as we still have `$(function(){})` and `$(document).ready()`, and while some may prefer `Promise.resolve(jQuery.ready)`, I imagine that `$(document).ready()` and `$(function() {})` will still be the most common, and a `document` context still makes sense to me on the latter 2. Like I said tho, I'm not opposed to changing that, but we don't have to change it now.